test/core/test/jest-expect.test.ts

```diff
@@ -51,6 +51,12 @@ describe('jest-expect', () => {
     expect(new Date(0)).toEqual(new Date(0))
     expect(new Date('inValId')).toEqual(new Date('inValId'))
 
+    expect(new Error('message')).toEqual(new Error('message'))
+    expect(new Error('message')).not.toEqual(new Error('different message'))
+
+    expect(new URL('https://example.org')).toEqual(new URL('https://example.org'))
+    expect(new URL('https://example.org')).not.toEqual(new URL('https://different-example.org'))
+
```

Comment:
Maybe also add a test case that includes search query parameters, just so that replacing `href` with other URL properties would fail these tests.

```suggestion
    expect(new URL('https://example.org?query=value')).toEqual(new URL('https://example.org?query=value'))
    expect(new URL('https://example.org?query=one')).not.toEqual(new URL('https://example.org?query=two'))

```

Reply:
Good idea. I’ve applied your suggestion.

Reply:
While we are at it, maybe we should also check hash and query+hash - different variations.

Reply:
I added two more sets of test cases:

- One set for URLs that contain a subdomain, path segment, query parameter, and fragment identifier
- One set for URLs using the `URL` constructor’s second argument for setting the base